### PR TITLE
smoke: add performance test

### PR DESCRIPTION
**Pull request by Desiki-high**
## Relevant Issue (if applicable)
#1460 

## Details
1. Remove old benchmark in `smoke test`.
2. Add a new performance test in `smoke test`, written in Golang to make it easy to maintain.
## Types of changes

_What types of changes does your PullRequest introduce? Put an `x` in all the boxes that apply:_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Go over all the following points, and put an `x` in all the boxes that apply._

- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

### CODE REVIEW

**Comment by Desiki-high** [comment]
PTAL. @imeoer

File: misc/performance/nydusd_config.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+            "type": "blobcache",
+            "config": {
+                "work_dir": "/var/lib/containerd-nydus/cache"
+            }
+        }
+    },
+    "mode": "direct",
+    "digest_validate": false,
+    "enable_xattr": true,
+    "iostats_files": true,
```
Comment (imeoer) [inline misc/performance/nydusd_config.json:20]: Let's set these metric options to false: `iostats_files`, `access_pattern`, `latest_read_files`.

File: misc/performance/prepare.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+readonly SNAPSHOOTER_VERSION=0.13.3
```
Comment (imeoer) [inline misc/performance/prepare.sh:3]: typo `SNAPSHOTTER_VERSION`.

File: smoke/Makefile [hunk 3]
```diff
@@ -1,6 +1,6 @@
 PACKAGES ?= $(shell go list ./... | grep -v /vendor/)
 GOPROXY ?= https://goproxy.io
-TESTS ?= .*
+TESTS ?= 'TestAPI|TestBlobCache|TestCompatibility|TestImage|TestNativeLayer|TestZranLayer'
```
Comment (imeoer) [inline smoke/Makefile:3]: Can we keep it with `.*`? Other developers to add test cases will forget to update this.

File: .github/workflows/smoke.yml [hunk 4]
```diff
@@ -677,3 +212,33 @@ jobs:
...
       - uses: actions/checkout@v3
       - uses: EmbarkStudios/cargo-deny-action@v1
+
+  performance-test:
+    runs-on: ubuntu-latest
+    needs: [contrib-build, nydus-build]
+    strategy:
+      matrix:
+        include:
+          - mode: nydus_v5
```
Comment (imeoer) [inline .github/workflows/smoke.yml:222]: How about `fs-version-5`/`fs-version-6`.

File: smoke/Makefile (hunk 3 above)
Comment (Desiki-high) [reply smoke/Makefile:3]: If we keep `.*`, when we use `TESTS` in `make test` we will run `performance_test` together. However, the performance test needs some env setup.

Comment (Desiki-high) [reply smoke/Makefile:3]: Maybe I can move the performance test to a separate directory in `smoke/tests` ?

File: misc/performance/prepare.sh (hunk 2 above)
Comment (Desiki-high) [reply misc/performance/prepare.sh:3]: fixed.

**Review by jiangliu — APPROVED** [review]

**Pull request merged**